pkg/gpu/stream.go

```diff
@@ -67,17 +80,68 @@ type kernelSpan struct {
 
 	// numKernels is the number of kernels that were launched during the span
 	numKernels uint64
+
+	// avgMemoryUsage is the average memory usage during the span, per allocation type
+	avgMemoryUsage map[model.MemAllocType]uint64
 }
 
-func newStreamHandler() *StreamHandler {
+func newStreamHandler(key streamKey, sysCtx *systemContext) *StreamHandler {
 	return &StreamHandler{
 		memAllocEvents: make(map[uint64]gpuebpf.CudaMemEvent),
+		key:            key,
+		sysCtx:         sysCtx,
 	}
 }
 
 func (sh *StreamHandler) handleKernelLaunch(event *gpuebpf.CudaKernelLaunch) {
-	// Copy events, as the memory can be overwritten in the ring buffer after the function returns
-	sh.kernelLaunches = append(sh.kernelLaunches, *event)
+	enrichedLaunch := &enrichedKernelLaunch{
+		CudaKernelLaunch: *event, // Copy events, as the memory can be overwritten in the ring buffer after the function returns
```

Comment:
do we need to keep them after the function returns? i am okay with a naive approach to start with, but I am pretty sure the performance overhead will be big once we stress test this module

Reply:
resolving for now.